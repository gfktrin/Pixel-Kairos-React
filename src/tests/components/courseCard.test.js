import React from 'react';
import { shallow } from 'enzyme';

import CourseCard from '../../components/courseCard/courseCard';

const props = {
  courseImg: 'https://site.com/link-to-image',
  courseName: 'Test Course',
  courseDescription: 'A course made for tests',
};

describe('CourseCard', () => {
  it('Tests default render with props', () => {
    const wrapper = shallow(
      <CourseCard {...props} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
