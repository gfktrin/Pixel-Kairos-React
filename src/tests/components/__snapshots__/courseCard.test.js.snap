// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCard Tests default render with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourseCard
    courseDescription="A course made for tests"
    courseImg="https://site.com/link-to-image"
    courseName="Test Course"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "actions": Array [
        <a
          className="course-link"
          href="#"
        >
          Conheça o curso
        </a>,
      ],
      "children": Array [
        <h5
          className="home-course-title"
        >
          Test Course
        </h5>,
        "A course made for tests",
      ],
      "className": "big",
      "header": <CardTitle
        image="https://site.com/link-to-image"
      />,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test Course",
          "className": "home-course-title",
        },
        "ref": null,
        "rendered": "Test Course",
        "type": "h5",
      },
      "A course made for tests",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "actions": Array [
          <a
            className="course-link"
            href="#"
          >
            Conheça o curso
          </a>,
        ],
        "children": Array [
          <h5
            className="home-course-title"
          >
            Test Course
          </h5>,
          "A course made for tests",
        ],
        "className": "big",
        "header": <CardTitle
          image="https://site.com/link-to-image"
        />,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Test Course",
            "className": "home-course-title",
          },
          "ref": null,
          "rendered": "Test Course",
          "type": "h5",
        },
        "A course made for tests",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
